kotlin-runtime/ftl-runtime/src/main/kotlin/xyz/block/ftl/logging/Logging.kt

```diff
@@ -29,18 +36,30 @@ class Logging {
       return logger(kClass.qualifiedName!!)
     }
 
-    fun init() {
-      val ple = PatternLayoutEncoder()
+    init {
+      val je = LoggingEventCompositeJsonEncoder()
+      je.context = logging.lc
 
-      ple.pattern = "%date %level %logger{10} - %msg%n"
-      ple.context = logging.lc
-      ple.start()
+      val providers: JsonProviders<ILoggingEvent> = je.providers
+      providers.setContext(je.context)
+      // Custom LogLevelJsonProvider converts level value to lowercase
+      providers.addProvider(object : LogLevelJsonProvider() {
+        override fun writeTo(generator: JsonGenerator, event: ILoggingEvent) {
+          JsonWritingUtils.writeStringField(generator, fieldName, event.level.toString().lowercase())
```

Comment:
How does it know the correct field names to use?

ie. we have

```json
{"message": "X", "error": "Y", "time": "Z", "attributes": {"..."}}
```

Reply:
the kotlin logger relies on the common log infrastructure to supplement "attributes" and "time" when the log is processed via the runner

for the log entry submitted to stderr - we're using the `JsonProviders` to customize the format. these specify field names "message", "error" and "level" and format their values as needed - so when these logs are written they look like `{"level": "<level", "message": "<message">, "error":"<error>"}` and can be parsed into an `Entry` object (and then modified to include additional fields)